packages/react-reconciler/src/ReactFiberNewContext.old.js

```diff
@@ -133,8 +133,14 @@ export function popProvider(
   pop(valueCursor, providerFiber);
   if (isPrimaryRenderer) {
     context._currentValue = currentValue;
+    if (__DEV__) {
+      context._currentRenderer = null;
```

Comment:
What I find odd is that ReactFizzNewContext does handle the `_currentRenderer` in `popProvider` but does not reset it  but set it it to `rendererSigil` i.e. push and pop do the same things to `_currentRenderer`:

https://github.com/facebook/react/blob/a724a3b578dce77d427bef313102a4d0e978d9b4/packages/react-server/src/ReactFizzNewContext.js#L273

Maybe we should actually restore `_currentRenderer` and not just reset it?

Reply:
> Maybe we should actually restore _currentRenderer and not just reset it?

Implemented in [`04c19a8` (#22797)](https://github.com/facebook/react/pull/22797/commits/04c19a81f8af947bb4f919a26082c1f642682415)